### ISSUE TITLE
[Bug] Tile layer issuing excess tile requests when map gets hidden

### ISSUE THREAD

**Issue by ilijapuaca** [visgl/deck.gl#7852]
### Description

When working with tile layer, hiding the entire map by setting the `display` style to `none` issues lots of new tile requests. 
This issue first manifested itself in the Jupyter notebook use case, but reproducing it in a React environment seems relatively easy too (I've pushed a trivial example on how to reproduce it to [this branch](https://github.com/visgl/deck.gl/tree/repro/tile-requests)).

Based on some light research, the problem seems to be that the viewport changes to quarter of its size when the DOM element gets hidden. Unsure whether this stems from some weird browser behavior where it emits updates with mixed device pixel sizes.

### Flavors

- [ ] Script tag
- [X] React
- [X] Python/Jupyter notebook
- [ ] MapboxOverlay
- [ ] GoogleMapsOverlay
- [ ] CartoLayer
- [ ] ArcGIS

### Expected Behavior

No tile requests get issued

### Steps to Reproduce

Start the `examples/website/map-tile` example on [this branch](https://github.com/visgl/deck.gl/tree/repro/tile-requests), open up the network tab and press the `Toggle Map` button

### Environment

- Framework version: 8.8.0, also on latest `master`
- Browser: Chrome 112.0.5615.137
- OS: macOS 13.3.1

**Comment by Pessimistress** [comment]
The issue with this code is that you are only hiding the canvas, not the element that receives pointer events, so the viewport will continue to update if you drag and scroll. To hide the React component you should do something like

```jsx
<DeckGL
  style={{display: showMap ? 'block' : 'none'}}
  ... />
```

**Comment by ilijapuaca** [comment]
I did not check what the `id` property of `DeckGL` gets assigned to under the hood, I assumed it was the top-level container but as you noted it seems to be the canvas.

I have revisited the example per your suggestion and pushed to the same branch, the same issue is still present. To make sure we're on the same page, there is no dragging or scrolling involved that causes the new tiles to be requested, I'm attaching a video just in case

https://user-images.githubusercontent.com/3406718/234706097-a7287998-d273-476c-9a0d-f5d259925855.mov

**Comment by Pessimistress** [comment]
Fixed in v8.9.12
